implementations/rust/ockam/ockam_api/src/kafka/protocol_aware/inlet/mod.rs

```diff
@@ -12,53 +13,78 @@ use std::sync::{Arc, Mutex};
 mod request;
 mod response;
 
+#[cfg(test)]
+mod tests;
+
 #[derive(Clone)]
 pub(crate) struct InletInterceptorImpl {
     request_map: Arc<Mutex<HashMap<CorrelationId, RequestInfo>>>,
     uuid_to_name: TopicUuidMap,
-    key_exchange_controller: KafkaKeyExchangeController,
+    key_exchange_controller: Arc<dyn KafkaKeyExchangeController>,
```

Comment:
I've tried to add `Clone` to it, but it can't be called using `KafkaKeyExchangeController` since it's being used as a dynamic Trait, and you can't return Self using a trait.